src/components/AddProductFiltersModal.vue

```diff
@@ -70,74 +76,77 @@ import {
 import { useStore } from "vuex";
-import { UtilService } from "@/services/UtilService";
 import { translate } from '@hotwax/dxp-components';
-import { hasError } from "@/utils";
 
 const queryString = ref('');
-const facetOptions = ref([]) as any;
-const isScrollable = ref(true);
 const selectedValues = ref([]) as any;
+const filteredOptions = ref([]) as any;
+const pageSize = process.env.VUE_APP_VIEW_SIZE || 0;
+const currentPage = ref(0);
+
+const isScrollable = ref(true);
+const isLoading = ref(false);
+const isScrollingEnabled = ref(false);
+const contentRef = ref({}) as any;
+const infiniteScrollRef = ref({}) as any;
 
 const props = defineProps(["label", "facetToSelect", "searchfield", "type"]);
 const store = useStore();
 
 const appliedFilters = computed(() => store.getters["util/getAppliedFilters"]);
+const facetOptions = computed(() => store.getters["util/getFacetOptions"]);
 
-onMounted(() => {
+onMounted(async() => {
+  isLoading.value = true;
+  await store.dispatch("util/fetchProductFilters", { facetToSelect: props.facetToSelect, searchfield: props.searchfield })
+  getFilters();
   selectedValues.value = JSON.parse(JSON.stringify(appliedFilters.value[props.type][props.searchfield]))
+  isLoading.value = false;
 })
 
 function closeModal() {
   modalController.dismiss({ dismissed: true });
 }
 
 function search() {
+  isScrollable.value = true;
+  currentPage.value = 0;
+  filteredOptions.value = []
+
   getFilters();
 }
 
-async function getFilters(vSize?: any, vIndex?: any) {
-  const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
-  const viewIndex = vIndex ? vIndex : 0;
-  
-  const payload = {
-    facetToSelect: props.facetToSelect,
-    docType: 'PRODUCT',
-    coreName: 'enterpriseSearch',
-    searchfield: props.searchfield,
-    jsonQuery: '{"query":"*:*","filter":["docType:PRODUCT"]}',
-    noConditionFind: 'N',
-    limit: viewSize,
-    q: queryString.value,
-    term: queryString.value,
-    offset: viewSize * viewIndex,
+async function loadMoreFilters(event: any){
+  // Added this check here as if added on infinite-scroll component the Loading content does not gets displayed
+  if(!(isScrollingEnabled.value && isScrollable.value)) {
+    await event.target.complete();
   }
 
-  const resp = await UtilService.fetchFacets(payload);
-  if(!hasError(resp)) {
-    const results = resp.data.facetResponse.response
-    facetOptions.value = viewIndex === 0 ? results : [...facetOptions.value , ...results];
-    isScrollable.value = (facetOptions.value.length % process.env.VUE_APP_VIEW_SIZE) === 0;
+  getFilters().then(() => {
+    event.target.complete();
+  })
+}
 
-  } else {
-    facetOptions.value = [];
-    isScrollable.value = false;
+async function getFilters() {
+  let options = facetOptions.value
+  if(queryString.value) {
+    options = facetOptions.value.filter((option: any) => option.label.toLowerCase().includes(queryString.value.toLowerCase()))
```

Comment:
We should think on improving this logic as this will also run on infinite scroll, and on each scroll we will first filter the records and then change the page. Instead we should update the logic to only filter the records once and change the page on infinite scroll.